gradle/instrumentation.gradle

```diff
@@ -23,24 +23,24 @@ if (projectDir.name == 'javaagent') {
 
 afterEvaluate {
   dependencies {
-    implementation project(':instrumentation-api')
-    implementation project(':javaagent-api')
+    compileOnly project(':instrumentation-api')
+    compileOnly project(':javaagent-api')
+    compileOnly project(':javaagent-bootstrap')
     // Apply common dependencies for instrumentation.
-    implementation(project(':javaagent-tooling')) {
+    compileOnly(project(':javaagent-tooling')) {
       // OpenTelemetry SDK is not needed for compilation, and :opentelemetry-sdk-shaded-for-testing
       // is brought in for tests by project(:testing-common) below
       exclude group: 'io.opentelemetry', module: 'opentelemetry-sdk'
     }
     implementation deps.bytebuddy
     annotationProcessor deps.autoservice
-    implementation deps.autoservice
+    compileOnly deps.autoservice
```

Comment:
This seemingly innocent change caused a lot of breakage since guava, previously set by autoservice, ended up getting too new for many of our projects. But it's probably better to have the manual downgrading as now to know about these instead of the luck.